hepdata/modules/records/utils/submission.py

```diff
@@ -570,6 +573,31 @@ def process_submission_directory(basepath, submission_file_path, recid,
 
         if len(errors) is 0:
             errors = package_submission(basepath, recid, hepsubmission)
+
+            # Check the size of the upload to ensure it can be converted
+            data_filepath = find_submission_data_file_path(hepsubmission)
+            with prepare_data_folder(data_filepath, 'yaml') as filepaths:
+                input_directory, input_file = filepaths
+                # Create options that look like a worst-case (biggest)
+                # conversions (using yoda-like options as they include rivet
+                # analysis
+                dummy_inspire_id = hepsubmission.inspire_id or '0000000'
+                options = {
+                    'input_format': 'yaml',
+                    'output_format': 'yoda',
+                    'filename': f'HEPData-ins{dummy_inspire_id}-v{hepsubmission.version}-yoda',
+                    'validator_schema_version': '0.1.0',
+                    'hepdata_doi': f'10.17182/hepdata.{recid}.v{hepsubmission.version}',
+                    'rivet_analysis_name': f'ATLAS_2020_I{dummy_inspire_id}'
+                }
+                data_size = get_data_size(data_filepath, options)
```

Comment:
Here `data_filepath` is the `.zip` file created by `package_submission`.  This is a bug and I think you mean `input_directory`.  Passing the `.zip` file (not a directory) to `get_data_size` means that [this line of hepdata-converter-ws-client](https://github.com/HEPData/hepdata-converter-ws-client/blob/144e8d477689368b78f0cf444c459dcdc00cd9f1/hepdata_converter_ws_client/__init__.py#L166) adds the `.zip` file to the tar file object without filtering to only `.yaml` or `.json` files.

It would be simpler just to call `data_size = get_data_size(basepath, options)` but I guess you want to use `prepare_data_folder` to mimic what's done in `process_zip_archive`.

Reply:
Oooops. Thought I'd traced it through and checked it was getting the right size back but apparently not.

> It would be simpler just to call data_size = get_data_size(basepath, options) but I guess you want to use prepare_data_folder to mimic what's done in process_zip_archive.

Yes, thought it best to do the same thing as the conversion. It does feel like we're over-complicating things somewhere along the line with all the zipping and unzipping, but that's a separate issue. (Would it be worth looking into though?)

Reply:
Yes, the use of `convert_zip_archive` in `download_submission` seems overly complicated:
https://github.com/HEPData/hepdata/blob/0528296cf3338850cbd32243bf348d098271f9df/hepdata/modules/converter/views.py#L314
It would be better just to pass the original directory of files to the converter, as done in `download_datatable`:
https://github.com/HEPData/hepdata/blob/0528296cf3338850cbd32243bf348d098271f9df/hepdata/modules/converter/views.py#L559-L566
I guess the problem is how to get the relevant directory in `download_submission`.  If you can work out how to do it, it might be worth doing.  The `convert_zip_archive` function is also used in `convert_endpoint`, but I've commented in the docstring that I don't think `convert_endpoint` is used anywhere so it could be removed.

Reply:
Given that the current system works despite the over-complication, should we log this as a new issue?

Reply:
OK, I'll log it as a new issue (low priority), given that it would require a significant refactoring.